pkgs/top-level/python-packages.nix

```diff
@@ -3882,6 +3882,10 @@ in {
 
   gpapi = callPackage ../development/python-modules/gpapi { };
 
+  gpaw = callPackage ../development/python-modules/gpaw {
+    fftw = pkgs.fftw.override { enableMpi = true; };
```

Comment:
If it's important that `gpaw` gets the *same* `fftw` as `libvdwxc` it might be an idea to add it to `libvdwxc`'s `passthru` and get it from there.

Reply:
Good suggestion, has been done :)

Reply:
Instead of using `passthru`, wouldn't it be better to put it `propgatedBuildInputs`?

Reply:
Hm, I am not sure about this one. The point here is that libvdwxc requires FFTW, not necessarily with MPI support on its own. GPAW however does require FFTW with MPI support and the libvdwxc that GPAW uses then also needs fftw with mpi support.

But an hypothetical other package (not that I am aware of one) that utilises libvdwxc that does not require fftw directly has no use for its propagated fftw.

Maybe a better alternative is to make MPI optional in libvdwxc 
```nix
{..., 
fftw,
mpi,
enableMpi ? true
}:

let fftwMpi = fftw.override { enableMpi = true; };

in stdenv.mkDerivation {
  ...
  buildInputs = [ ... ] ++ if enableMpi then [fftwMpi mpi] else [fftw];
}
```

and in gpaw
```nix
{..., 
fftw,
libvdwxc
}:
let libvdwxcMpi = libvdwxc.override { enableMpi = true; };
    fftwMpi = fftw.override { enableMpi = true; };
...
```

Also not optimal but maybe better?

Reply:
I think that `propagatedBuildInputs` could be the simpler solution here(?):
* Yes, a package that uses `libvdwxc` does not need `fftw` will be offered `fftw`. However, this should not hurt either.
* With `fftwMpi` in `propagatedBuildInputs` you can even remove `fftw` altogether from the `gpaw` inputs, and it is guaranteed that `gpaw` will get the right `fftw` version.
* Using `pasthru` is more complex: anyone building a new package needs to lookup `libvdwxc`'s passthru flags to get the `fftw` right.

Btw.: I think we can even define a top-level `fftwMpi` attribute now.

Reply:
>  I think we can even define a top-level fftwMpi attribute now.

I have added one :)

I guess multiple options are somehow valid. For now I've implemented the `propagatedBuildInputs` version.